### PR TITLE
Modularize Crew Monitor, fixes health status

### PR DESCRIPTION
**Pull request by Zergspower**
## About The Pull Request

With TG Mirrors a real probability here, i've added CrewConsoleSkyrat.js to account for this

## How This Contributes To The Skyrat Roleplay Experience

Better represents the health of players who otherwise would be in peril, this would allow medical or other parties to better gauge if someone is dead or in critical in a more correct manner. Previously alive crew were showing up dead, this clearly was a cause of stress when a paramedic now assumes you are the walking dead.

## Changelog

image

Readdresses the math relating to the status icons that haven't been correct in quite a while, given the healthpool differences we were running off TG's formula. As you 'die' when you hit 200 damage I ensured the critical icon doesn't show until you're actually in critical now rather than the 100-125 zone.

:cl:
qol: Icons now correctly indicate health status
fix: Fixes the maths relating to the health pool, no longer working off 1-125
code: migrated CrewConsole to a Skyrat Modular file to avoid future clashes with mirrors
/:cl:

### CODE REVIEW

**Comment by Zergspower** [comment]
i've no idea why CI Failed if anyone can explain that

**Comment by Zonespace27** [comment]
> i've no idea why CI Failed if anyone can explain that

run prettierjs
image

File: tgui/packages/tgui/interfaces/CrewConsoleSkyrat.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+      inline
+      width={2}
+      color={COLORS.damageType[type]}
+      textAlign="center">
+      {value}
+    </Box>
+  );
+};
+
+export const CrewConsole = () => {
```
Comment (Zonespace27) [inline tgui/packages/tgui/interfaces/CrewConsoleSkyrat.js]: make this a skyrat version - overrides don't work in js/ts/tsx to my knowledge

File: tgui/packages/tgui/interfaces/CrewConsoleSkyrat.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
+import { sortBy } from 'common/collections';
```
Comment (Zonespace27) [inline tgui/packages/tgui/interfaces/CrewConsoleSkyrat.js:1]: Replace the UI in the crew monitor file with this UI

File: tgui/packages/tgui/interfaces/CrewConsoleSkyrat.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+import { useBackend } from '../backend';
```
Comment (Zonespace27) [inline tgui/packages/tgui/interfaces/CrewConsoleSkyrat.js:2]: Reset the non-SR version of this file

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/.prettierrc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (Zonespace27) [inline tgui/packages/tgui/interfaces/.prettierrc]: get rid of this
(also pull from master to be safe)

**Comment by Zonespace27** [comment]
Ping a maintainer to reopen when changes are made

**Comment by Zergspower** [comment]
@Zonespace27

**Review by GoldenAlpharex — APPROVED** [review]
Looks good!

File: tgui/packages/tgui/interfaces/CrewConsoleSkyrat.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    <Table.Row>
+      <Table.Cell bold={jobIsHead(ijob)} color={jobToColor(ijob)}>
+        {name}
+        {assignment !== undefined ? ` (${assignment})` : ''}
+      </Table.Cell>
+      <Table.Cell collapsing textAlign="center">
+        {is_robot ? <Icon name="wrench" color="#B7410E" size={1} /> : ''}
+      </Table.Cell>
+      <Table.Cell collapsing textAlign="center">
+        {oxydam !== undefined ? (
```
Comment (GoldenAlpharex) [inline tgui/packages/tgui/interfaces/CrewConsoleSkyrat.js]: This should make it so it displays a skull icon when they're dead, even if their health is all fine and dandy. Do make sure to test it before committing it.
```suggestion
        {oxydam !== undefined && life_status ? (
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
My final little nitpick.

**Comment by Zergspower** [comment]
image
 Fascinating

**Review by GoldenAlpharex — APPROVED** [review]
Alright, it's all good to go now. Thanks for all your hard work on this!

**Pull request merged**